src/EntityFramework/Core/Common/DbProviderServices.cs

```diff
@@ -863,5 +866,16 @@ public virtual IEnumerable<object> GetServices(Type type, object key)
         {
             return _resolvers.GetServices(type, key);
         }
+
+        /// <summary>
+        /// By default expression like .Select(x => NewProperty = func(x.Property)).Where(x => x.NewProperty == ...) are simplified to avoid nested SELECT
```

Comment:
Can't remember what the width limit is we are trying to follow in this codebase but this looks longer. Please word wrap.

Reply:
Done. However, it's in different fil now